### ISSUE TITLE
Network State Unkmown Error

### ISSUE THREAD

**Issue by moringaman** [loomnetwork/truffle-dappchain-example#3]
I am trying to run the example but receive the following error when I run yarn deploy.

```bash
yarn deploy
yarn run v1.9.4
$ truffle deploy --network loom_dapp_chain
Using network 'loom_dapp_chain'.

Running migration: 1_initial_migration.js
  Deploying Migrations...
Error encountered, bailing. Network state unknown. Review successful transactions manually.
Network Error
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

I am running the loom node on the same machine the output of which is

```bash 
I[08-04|23:43:11.543] Using simple log event dispatcher            
I[08-04|23:43:11.567] Deployed contract                            vm=plugin location=coin:1.0.0 name=coin address=default:0xe288d6eec7150D6a22FDE33F0AA2d81E06591C4d
Init DPOS Params &dpos.DPOSInitRequest{Params:(*dpos.Params)(0xc420b7e6f0), Validators:[]*types.Validator{(*types.Validator)(0xc420b8da20)}}
I[08-04|23:43:11.567] Deployed contract                            vm=plugin location=dpos:1.0.0 name=dpos address=default:0x01D10029c253fA02D76188b84b5846ab3D19510D
```

I have tried this both on linux and OSX with exactly the same results. Please advise

Thnx

**Comment by eduardonunesp** [comment]
Hey @moringaman this error generally happens when the `truffle` isn't capable of reach the address defined on the `truffle.js` configuration. If you're using the default configuration should be `127.0.0.1:46658`. Could you please check if after run the `./loom` program if the port is available. You can check running `lsof -i -n -P | grep -i 46658`, if the host and port are returned then you can please run `curl localhost:46658/rpc` to check if you can reach the Loomchain.

**Comment by moringaman** [comment]
Hi @eduardonunesp Thanks for the response to this one, I have run the checks you have suggested and am not getting the loom rpc service listening on port 46658. Have followed the instructions to the letter althought they do seem very concise.

One thins I would say is that the instructions for setting up and running loom on a linux (deb) based system given on this url: https://loomx.io/developers/docs/en/prereqs-ubuntu.html are a lot different to the ones at this ur:l https://loomx.io/developers/docs/en/basic-install-linux.html.

I am still a little unsure however what the full procedure for installing and running a local loom node are?

P.S I've tried this on Arch, Lubuntu and OSX with the same results

**Comment by eduardonunesp** [comment]
@moringaman thanks for the report, could you please inform what is the current version used? `./loom version`

**Comment by moringaman** [comment]
@eduardonunesp The version is 1.0.0+b330

**Comment by moringaman** [comment]
No loom.yaml, does that get created with ./loom init ?

**Comment by eduardonunesp** [comment]
Nope, but you can try to create a file `loom.yaml` on the same directory and try to set the config `RPCProxyPort: 46658` and make sure that config on `truffle-dappchain-example`  uses the port `46658` too

**Comment by moringaman** [comment]
I created a loom.yaml file in the same directory as the loom binary and checked truffle.js which is referencing port 46658 but I'm still getting the same result. Can you confirm what I should expect to see in the console when I run ./loom run?

**Comment by eduardonunesp** [comment]
I just tested here, with the latest `truffle-dappchain-example` and `loom`  build `330`, and you should see something like: 

```bash
$ yarn deploy:reset
Using network 'loom_dapp_chain'.

Running migration: 1_initial_migration.js
  Replacing Migrations...
  ... 0x02666a3bea90d1c4fd6420cc548c3654b1f871355326b1e8c72be1cda4665e4a
  Migrations: 0x6d002528e54a421f5e7375690e5fb74b75add0d8
Saving successful migration to network...
  ... 0x0653423baa7994fe3a5eb6712382f3a783478180488f2b83359627900297436b
Saving artifacts...
Running migration: 2_simple_store.js
  Replacing SimpleStore...
  ... 0x87cefb629176a846122eed354522f003b4d865e3e3274be880a47abb7265213f
  SimpleStore: 0x1a31b9b9d281d49001fe7f3f638000a739afc9c3
Saving successful migration to network...
  ... 0x7ddd8df83539848f21df3908fc4e24a630b53a82235b8a31ad9e03c8bc99c36d
Saving artifacts...
```
Also testing
```bash
yarn test
yarn run v1.7.0
$ truffle test --network loom_dapp_chain
Using network 'loom_dapp_chain'.



  Contract: SimpleStore
    ✓ Should have an address for SimpleStore
    ✓ Should set a value (1019ms)


  2 passing (5s)
```

Running on `macOS 10.13.6`
Go version `go1.10.2 darwin/amd64`
Node verison  `v10.7.0`

**Comment by moringaman** [comment]
And when you execute ./loom run? what do you see in the console?

**Comment by eduardonunesp** [comment]
Humm, interesting, maybe there's some issue when you try to map to `localhost (127.0.0.1)`

**Comment by moringaman** [comment]
I have tried this on a number of systems (3) to be exact by following the online guide without any success so maybe the guide is missing a few steps. I am able to run a geth node locally mapping to localhost 127.0.0.1 without any problems.

Also why is there no mention of the docker image in the docks and do you know whether it's being maintained as I can see that it's running loom build 1.0.0+b349 which is ahead of the one in the example?

**Comment by mattkanwisher** [comment]
oh the docker image is probably to new, you likely want to run build 330

**Comment by moringaman** [comment]
So theoretically one could just copy the build 330 binary into the docker container and run it?

Also for use with the truffle-dappchain-example it is necessary to expose all the websocket ports in the run command i.e.

`` docker run -p 46656:46656 -p 46657:46657 -p 46658:46658 -p 9999:9999 loomnetwork/loom ``

**Comment by eduardonunesp** [comment]
This example is update with latest loom SDK

**Comment by eduardonunesp** [comment]
Tested recently and Loom SDK with Docker and just works, I'll close this issue due to the inactivity also
